client/state/data-layer/wpcom/sites/plugins/new/index.js

```diff
@@ -21,6 +28,38 @@ export const uploadPlugin = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
+const showSuccessNotice = ( dispatch, { name } ) => {
+	dispatch( successNotice(
+		translate( "You've successfully uploaded the %(name)s plugin.", {
+			args: { name }
+		} ),
+		{ duration: 5000 }
+	) );
+};
+
+const showErrorNotice = ( dispatch, error ) => {
+	const knownErrors = {
+		exists: translate( 'This plugin is already installed on your site.' ),
+		'too large': translate( 'The plugin zip file must be smaller than 10MB.' ),
+		incompatible: translate( 'Not a compatible plugin.' ),
+		unsupported_mime_type: translate( 'Not a valid zip file.' ),
+	};
+	const errorString = toLower( error.error + error.message );
+	const knownError = find( knownErrors, ( v, key ) => includes( errorString, key ) );
+
+	if ( knownError ) {
+		dispatch( errorNotice( knownError ) );
+		return;
+	}
+	if ( error.error ) {
+		dispatch( errorNotice( translate( 'Upload problem: %(error)s.', {
+			args: { error: error.error }
+		} ) ) );
+		return;
+	}
+	dispatch( errorNotice( translate( 'Problem installing the plugin.' ) ) );
```

Comment:
Hi! I've found a possible matching string that has already been translated **38** times:
`translate( 'Installing the plugin.' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).